chainer/functions/array/stack.py

```diff
@@ -6,12 +6,61 @@ def stack(xs, axis=0):
     """Concatenate variables along a new axis.
 
     Args:
-        xs (list of chainer.Variable): Variables to be concatenated.
-        axis (int): Axis of result along which variables are stacked.
+        xs (list of :class:`~chainer.Variable` or :class:`numpy.ndarray` or \
+        :class:`cupy.ndarray`):
+            Input variables to be concatenated. The variables must have the
+            same shape.
+        axis (int): The axis along which the arrays will be stacked. Default
+            is 0.
```

Comment:
Thank you for explanation. Then, could you write this restriction to the description of `axis`? Aside from that, I noticed that we do not check `axis=-(ndim + 1)` case in the unit test. I will add the case as another PR.